File: engine/src/com/magma/engine/entities/StatusEffect.java
package com.magma.engine.entities;

public class StatusEffect {	
	public int maxHealth;
	public int maxEnergy;
	public int maxFear;
	
	public int attack;
	public int defense;
	public int luck;
	public int speed;
	
	public StatusEffect add(StatusEffect with) {
		maxHealth += with.maxHealth;
		maxEnergy += with.maxEnergy;
		maxFear += with.maxFear;
		attack += with.attack;
		defense += with.defense;
		luck += with.luck;
		maxFear += with.maxFear;
		speed += with.speed;
		return this;
	}
	
	public StatusEffect subtract(StatusEffect with) {
		maxHealth -= with.maxHealth;
		maxEnergy -= with.maxEnergy;
		maxFear -= with.maxFear;
		attack -= with.attack;
		defense -= with.defense;
		luck -= with.luck;
		maxFear -= with.maxFear;
		speed -= with.speed;
		return this;
	}
	
	public static StatusEffect invincible() {
		final int MAX = 9999;
		StatusEffect stat = new StatusEffect();
		stat.maxHealth = MAX;
		stat.maxEnergy = MAX;
		stat.maxFear = MAX;
		stat.attack = MAX;
		stat.defense = MAX;
		stat.luck = MAX;
		stat.maxFear = MAX;
		stat.speed = 4;
		return stat;
	}

	@Override
	public String toString() {
		return "EN=" + maxHealth + ", CA=" + maxEnergy + ", NU=" + maxFear
				+ ", AT=" + attack + ", DEF=" + defense + ", LU=" + luck + ",SP="
				+ speed;
	}
}

File: engine/src/com/magma/engine/entities/StatusEffect.kt
package com.magma.engine.entities

class StatusEffect {
    var maxHealth = 0
    var maxEnergy = 0
    var maxFear = 0
    var attack = 0
    var defense = 0
    var luck = 0
    var speed = 0

    fun add(with: StatusEffect): StatusEffect {
        maxHealth += with.maxHealth
        maxEnergy += with.maxEnergy
        maxFear += with.maxFear
        attack += with.attack
        defense += with.defense
        luck += with.luck
        maxFear += with.maxFear
        speed += with.speed
        return this
    }

    fun subtract(with: StatusEffect): StatusEffect {
        maxHealth -= with.maxHealth
        maxEnergy -= with.maxEnergy
        maxFear -= with.maxFear
        attack -= with.attack
        defense -= with.defense
        luck -= with.luck
        maxFear -= with.maxFear
        speed -= with.speed
        return this
    }

    override fun toString(): String {
        return ("EN=" + maxHealth + ", CA=" + maxEnergy + ", NU=" + maxFear
                + ", AT=" + attack + ", DEF=" + defense + ", LU=" + luck + ",SP="
                + speed)
    }

    companion object {
        val invincible: StatusEffect
            get() {
                val max = 9999
                val stat = StatusEffect()
                stat.maxHealth = max
                stat.maxEnergy = max
                stat.maxFear = max
                stat.attack = max
                stat.defense = max
                stat.luck = max
                stat.maxFear = max
                stat.speed = 4
                return stat
            }
    }
}